### DISCUSSION TITLE
Metadata load triggered from SQL Editor

### DISCUSSION THREAD

**Discussion by alonpr** [dbeaver/dbeaver#22382] [DBeaver Development]
I've noticed that some text editing actions (couldn't reproduce which ones exactly but it happens rather frequently), seems like marking a text or copy-paste in a connected SQL script editor window trigger a metadata load related to the text by the cursor.
For example, with a connection to Snowflake:

![DBeaver](https://github.com/dbeaver/dbeaver/assets/44065703/09f31f88-6ecc-4269-8b6f-eed0d80499a4)

![SF](https://github.com/dbeaver/dbeaver/assets/44065703/cce3bc82-da12-41bb-86c2-65ecf63736cf)

Is it that some keys sequence triggering auto complete or something of that nature that requires a metadata load?
I checked the setting and seems like I have all the sql completion turned off:

![Preferences](https://github.com/dbeaver/dbeaver/assets/44065703/c30d83db-c406-46f6-b8c6-335ef5db2bbf)

**Comment by LonwoLonwo** [comment]
Hello @alonpr

You need to check these settings as well:

![2024-01-09 11_26_53-Preferences](https://github.com/dbeaver/dbeaver/assets/45152336/501ca195-626c-4a37-8e8f-366befd1d1e8)

**Reply by alonpr** [reply]
I had only the "Read database table/columns for semantic analysis" checked, I un-checked it but these calls still happen.

**Comment by alonpr** [comment]
It's happening when I press (hold) the "Command" key (macOS) and hover the mouse cursor over some object names in the SQL editor.

**Comment by alonpr** [comment]
I unchecked this as well but it's still happening:

![image](https://github.com/dbeaver/dbeaver/assets/44065703/b714a5f2-f4dd-48e2-93b1-4790f73141f7)

**Comment by alonpr** [comment]
It seems related to hyperlinking, I un-checked it in settings but it's still active, tried to close and re-open the app but it didn't help.

**Comment by alonpr** [comment]
Can anyone take a look and advice?
In Snowflake, as probably with any other on-demand paid database solution, it might be costly that command are issued in the background without means to control or disable it.

**Comment by alonpr** [comment]
@LonwoLonwo any idea?

**Comment by Matvey16** [comment]
Seems like a bug, for me it reproduced vut with an "option" key instead

**Reply by alonpr** [reply]
@Matvey16 thanks for checking it. Do you mind filing a bug for it?

**Reply by Matvey16** [reply]
YEah, i'll open a bug on our board

**Reply by LonwoLonwo** [reply]
I already did it @Matvey16

**Reply by LonwoLonwo** [reply]
https://github.com/dbeaver/dbeaver/issues/22555
